index.bs

```diff
@@ -402,9 +567,27 @@ perform the following steps:
 
 </div>
 
-Every [=bounce tracking timer period=] the user agent should run the
-[=bounce tracking timer=] algorithm given the [=wall clock=]'s
-[=wall clock/unsafe current time=].
+The user agent should run the [=record stateful bounces for bounce tracking=] algorithm once,
+[=client bounce detection timer period=] after the [=bounce tracking record/client bounce detection timer=]
```

Comment:
I see how you modeled this after the other time, but I think they are a bit different.  The other timer is a global timer that runs on a fixed interval.

I think what we want here is more like how `setTimeout()` is specified.  It uses `run steps after a timeout`:

https://html.spec.whatwg.org/#run-steps-after-a-timeout

You just have to be careful not to pass in a timerkey param so you don't conflict with setTimeout().

For completionSteps you would pass your algorithm `given |navigable|`.

To handle canceling I think we could make the completion steps abort if the navigable has changed.  For example, at start of navigation set a new timer key on the navigable, pass the timer key as part of completion steps, and then verify they haven;t changed when completionSteps runs.

@jyasskin WDYT?

Reply:
I haven't read this section yet, but is https://infra.spec.whatwg.org/#abort-when what you want?

Reply:
Thanks for the link, Ben. I think the `run steps after a timeout` algorithm is what I'm looking for. I'm not sure if `abort when` works here, because we want the callback to run after the timer, rather than before.

I added a boolean "running" flag on the navigable to handle cancelling. (Are there thread-safety concerns which need to be addressed or is that an implementation detail?) Leaving a short blurb here to explain when this algorithm is run, but the actual calls to it are located throughout the spec.